ocaml/libs/uuid/uuidx.mli

```diff
@@ -27,14 +27,26 @@
 type 'a t
 
 val null : 'a t
-(** A null UUID, as if such a thing actually existed. It turns out to be
-    useful though. *)
+(** A null UUID, as defined in RFC 9562 5.9. *)
 
 val make : unit -> 'a t
 (** Create a fresh UUID *)
 
 val make_uuid_urnd : unit -> 'a t
 
+val make_v4_uuid : int64 -> int64 -> 'a t
+(** For testing only: Create a v4 UUID, as defined in RFC 9562 5.4 *)
+
+val make_v7_uuid_from_time_and_bytes : int * int64 -> int64 -> 'a t
+(** For testing only: create a v7 UUID, as defined in RFC 9562 5.7 *)
+
+val make_v7_uuid : unit -> 'a t
```

Comment:
Could this be changed to accept a function for the timestamp? That way we could use a monotonic clock source. The signature could be `(unit -> uint64) -> 'a t`
I know it would not be standard-compliant, and we would need to be careful about not persisting the produced timestamps, but I believe losing monotonicity here is something very undesired.

Reply:
I've gone for a hybrid approach whereby I use ptime to get the system time initially, then use mtime to get monotonic adjustments to that - so that we keep monotonicity but still have timestamps that are at least approximately correct.